woodwork/table_accessor.py

```diff
@@ -248,6 +248,34 @@ def _get_typing_info(self):
         typing_info = typing_info.loc[list(self._dataframe.columns), :]
         return typing_info
 
+    @property
+    def name(self):
+        """Name of the DataFrame"""
+        if self._schema is None:
+            _raise_init_error()
+        return self._schema.name
+
+    @name.setter
+    def name(self, new_name):
+        """Set name of the DataFrame"""
+        if self._schema is None:
+            _raise_init_error()
+        self._schema.name = new_name
+
+    @property
+    def metadata(self):
```

Comment:
`name` and `metadata` should probably be added to the api reference for WoodworkTableAccessor